src/server.ts

```diff
@@ -1,97 +1,246 @@
+const file_proto : any = protoDescriptor.file;
+
+export const serviceNames: string[] = ['', 'file.fileService'];
+export const healthCheckStatusMap = {
+  '': HealthCheckResponse.ServingStatus.UNKNOWN,
+  serviceName: HealthCheckResponse.ServingStatus.UNKNOWN
+};
+
+/**
+ * The FileServer class, containing all of the FileServer methods.
+ */
+export class FileServer {
+
+  public server: grpc.Server;
+  public grpcHealthCheck: GrpcHealthCheck;
+
+  public constructor(port: string) {
+    this.server = new grpc.Server();
+    this.addServices();
+    this.server.bind(`0.0.0.0:${port}`, grpc.ServerCredentials.createInsecure());
+    log(Severity.INFO, 'server bind', `server listening on port: ${port}`);
+  }
+
+  private addServices () {
+    // Register the health service
+    this.grpcHealthCheck = new GrpcHealthCheck(healthCheckStatusMap);
+    this.server.addService(HealthService, this.grpcHealthCheck);
+
+    const fileService = {
+      GenerateKey: this.wrapper(this.GenerateKey),
+      CreateUpload: this.wrapper(this.CreateUpload),
+      UpdateUploadID: this.wrapper(this.UpdateUploadID),
+      GetUploadByID: this.wrapper(this.GetUploadByID),
+      DeleteUploadByID: this.wrapper(this.DeleteUploadByID),
+      GetFileByID: this.wrapper(this.GetFileByID),
+      GetFileByKey: this.wrapper(this.GetFileByKey),
+      GetFilesByFolder: this.wrapper(this.GetFilesByFolder),
+      CreateFile: this.wrapper(this.CreateFile),
+      DeleteFile: this.wrapper(this.DeleteFile),
+      IsAllowed: this.wrapper(this.IsAllowed),
+    };
+
+    this.server.addService(file_proto.FileService.service, fileService);
+  }
+  /**
+   * wraps all of the service methods, creating the transaction for the apm and the logger.
+   * @param func - the method called.
+   */
+  private wrapper (func: Function) :
+  (call: grpc.ServerUnaryCall<Object>, callback: grpc.requestCallback<Object>) => Promise<void> {
+    return async (call: grpc.ServerUnaryCall<Object>, callback: grpc.requestCallback<Object>) => {
+      try {
+        const traceparent = call.metadata.get('elastic-apm-traceparent');
+        const transOptions = (traceparent.length > 0) ? { childOf: traceparent[0].toString() } : {};
+        apm.startTransaction(`/file.FileService/${func.name}`, 'request', transOptions);
+        const traceID: string = getCurrTraceId();
+        log(Severity.INFO, func.name, 'request', traceID, call.request);
+
+        const res = await func(call, callback);
+
+        apm.endTransaction(statusToString(grpc.status.OK));
+        log(Severity.INFO, func.name, 'response', traceID, res);
+        callback(null, res);
+      } catch (err) {
+        const validatedErr : ApplicationError = validateGrpcError(err);
+        log(Severity.ERROR, func.name, err.message, getCurrTraceId());
+        apm.endTransaction(validatedErr.name);
+        callback(validatedErr);
+      }
+    };
+
+  }
+
+  // ******************** UPLOAD FUNCTIONS ******************** */
```

Comment:
Why not to seperate to diffrent files (for upload and file).

Reply:
done. Called it file.grpc and upload.grpc.
If you have a better name, I am open to suggestions